core/src/main/java/org/apache/druid/data/input/IntermediateRowParsingReader.java

```diff
@@ -93,51 +111,126 @@ public InputRow next()
       @Override
       public void close() throws IOException
       {
-        intermediateRowIterator.close();
+        intermediateRowIteratorWithMetadata.close();
       }
     };
   }
 
   @Override
   public CloseableIterator<InputRowListPlusRawValues> sample() throws IOException
   {
-    return intermediateRowIterator().map(row -> {
 
-      final List<Map<String, Object>> rawColumnsList;
-      try {
-        rawColumnsList = toMap(row);
-      }
-      catch (Exception e) {
-        return InputRowListPlusRawValues.of(null,
-                                            new ParseException(String.valueOf(row), e, "Unable to parse row [%s] into JSON", row));
-      }
+    final CloseableIteratorWithMetadata<T> delegate = intermediateRowIteratorWithMetadata();
+    final BiFunction<T, Map<String, Object>, InputRowListPlusRawValues> samplingFunction =
+        (row, metadata) -> {
```

Comment:
Instead of this big lambda, can you create a static method and pass the name of the method here in lambda expression here.

Reply:
The function uses `toMap` which is overridden in the subclasses, so creating a new non-static method